### ISSUE TITLE
No wifi on Lenovo Yoga 6 AMD Ryzen 5700 with rtl8852AE

### ISSUE THREAD

**Issue by easlingml** [pop-os/pop#1893]
**Distribution (run `cat /etc/os-release`):**
NAME="Pop!_OS"
VERSION="21.04"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.04"
VERSION_ID="21.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute
LOGO=distributor-logo-pop-os

**(run 'lspci'):**

0:00.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Root Complex
00:00.2 IOMMU: Advanced Micro Devices, Inc. [AMD] Renoir IOMMU
00:01.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge
00:02.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge
00:02.1 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe GPP Bridge
00:02.4 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe GPP Bridge
00:08.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge
00:08.1 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir Internal PCIe GPP Bridge to Bus
00:08.2 PCI bridge: Advanced Micro Devices, Inc. [AMD] Renoir Internal PCIe GPP Bridge to Bus
00:14.0 SMBus: Advanced Micro Devices, Inc. [AMD] FCH SMBus Controller (rev 51)
00:14.3 ISA bridge: Advanced Micro Devices, Inc. [AMD] FCH LPC Bridge (rev 51)
00:18.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 0
00:18.1 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 1
00:18.2 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 2
00:18.3 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 3
00:18.4 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 4
00:18.5 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 5
00:18.6 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 6
00:18.7 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir Device 24: Function 7
01:00.0 Non-Volatile memory controller: Micron/Crucial Technology Device 5407
02:00.0 Network controller: Realtek Semiconductor Co., Ltd. Device 8852
03:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Lucienne (rev c1)
03:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Device 1637
03:00.2 Encryption controller: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) Platform Security Processor
03:00.3 USB controller: Advanced Micro Devices, Inc. [AMD] Renoir USB 3.1
03:00.4 USB controller: Advanced Micro Devices, Inc. [AMD] Renoir USB 3.1
03:00.5 Multimedia controller: Advanced Micro Devices, Inc. [AMD] Raven/Raven2/FireFlight/Renoir Audio Processor (rev 01)
03:00.6 Audio device: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) HD Audio Controller
04:00.0 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 81)
04:00.1 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 81)

**Issue/Bug Description:**
 PopOS doesn't recognize onboard wifi adapter - RTL8852AE (FCC ID: TX2-R8852AE,  FRU: 5W10V25797).
 PCLinuxOS does recognize the wifi adapter perfectly - it's using the 5.8 kernel. Ergo, I know the device name listed above.
 If I plug in a USB wifi adapter, PopOS sees and uses it correctly. Also, networking works correctly if I plug in a USB-C network cable adapter connected directly to my router.

I tried running the solution provided in " Hp envy x360 ryzen 5500u realtek wifi 2x2 driver issue #1834," but to no avail. Got several errors. One of which said aborting due to missing VMLinux.

HELP.

 

**Steps to reproduce (if you know):**

Either booting from Live USB or after installing to metal, NO OnBoard WIFI.

**Expected behavior:**

Wifi should work right out of the gate just as in PCLinuxOS - which I don't like. In fact, PCLinux OS is the ONLY distro I tried, debian or arch based. I tried SparkyLinux, EndeavourOS, Manjaro, MX-Linux, KDE Neon, LMDE4, Debian 11 with non-free firmware, elementary, Solus, Bohdi, Garuda, KaOS, Fedora Workstation, Sabayon, CrunchBang++ (!#++), and more. None, other than PCLinuxOS, recognized the onboard wifi.

**Other Notes:**

**Comment by easlingml** [comment]
Fixed the issue.

 Sorry to have bothered all of you for my juvenile stupidity. 

I want to give a big thanks to "Hp envy x360 ryzen 5500u realtek wifi 2x2 driver issue #1834". After having read that issue, which led me to the lwfinger / rtw89 posting, I knew there should be a fix; and there was.  But not immediately - due to my juvenile stupidity. 

I tried the rtw89 "fix" not once, not twice, but thrice. All to no avail. I kept getting certain errors. Finally, after having read the rtw98 posting again, and again, and again, I found the errors of my ways. A typo I kept making over and over. Once I corrected for my typing error, everything went thru without a glitch.

And, Viola!!!

I now have onboard rtl8852ae wifi card recognition & connectivity. Thank goodness. I was just getting short of buying a Dell Inspiron which has onbaord wifi recognition without any kernel mods right out of the box. But I like this Lenovo Yoga 6 with AMD Ryzen 5700 cpu soooooo much. It's snappier and more responsive than the Dell, IMHO. 

I am free to explore other Debian Based Distros once again. 

Now, if we could only get FireFox to respond to touchscreen controls, pinch-to-zoom for one, like Chrome & Chromium do. . . .
Well, I guess that's another story.

MLE
